library/Icingadb/Common/Macro.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+/* Icinga DB Web | (c) 2021 Icinga GmbH | GPLv2 */
+
+namespace Icinga\Module\Icingadb\Common;
+
+use Icinga\Application\Logger;
+use Icinga\Module\Icingadb\Compat\CompatHost;
+use Icinga\Module\Icingadb\Compat\CompatService;
+
+trait Macro
+{
+    /**
+     * Known icinga macros
+     *
+     * @var string[]
+     */
+    protected $icingaMacros = [
+        'HOSTNAME'            => 'host_name',
+        'HOSTADDRESS'         => 'host_address',
+        'HOSTADDRESS6'        => 'host_address6',
+        'SERVICEDESC'         => 'service_description',
+        'host.name'           => 'host_name',
+        'host.address'        => 'host_address',
+        'host.address6'       => 'host_address6',
+        'service.description' => 'service_description',
+        'service.name'        => 'service_description'
```

Comment:
Our native column expressions are separated by dots. There's no `host_name`, just a `host.name`. We should not use different identifiers.

Reply:
The expressions the user has to specify are still separated by dots, but the actual `LegacyColumns` are not separated in any way by dot. See [here](https://github.com/Icinga/icingadb-web/blob/master/library/Icingadb/Compat/CompatHost.php#L13).

Reply:
Never mind.

Reply:
We will then have to adjust the `CompatObject`'s `__get()` method because it only matches an e.g. `service|host_propertie`.